File: datastructure&algorithm/ds_algo_java/src/daily/y_2022/m_11/w_02/d_2022_11_09/bj_1377_3/Main.java
package daily.y_2022.m_11.w_02.d_2022_11_09.bj_1377_3;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static void main(String[] args) throws IOException {
        int n = Integer.parseInt(br.readLine());
        Node[] arr = new Node[n];

        for(int i=0;i<n;i++){
            Node node = new Node(Integer.parseInt(br.readLine()),i);
            arr[i] = node;
        }
        Arrays.sort(arr);

        int max = Integer.MIN_VALUE;
        for(int i=0;i<n;i++){
            max = Math.max(arr[i].idx-i,max);

        }
        System.out.println(max+1);
    }

    static class Node implements Comparable<Node>{
        int val;
        int idx;
        public Node(int val, int idx){
            this.val = val;
            this.idx = idx;
        }

        public int compareTo(Node o){
            return val-o.val;
        }
    }
}

File: datastructure&algorithm/ds_algo_java/src/daily/y_2022/m_11/w_02/d_2022_11_09/bj_1377_3/Main.kt
package daily.y_2022.m_11.w_02.d_2022_11_09.bj_1377_3
import algorithms.sort.bubble_sort.bj_1377_3.Node
import java.io.BufferedReader;
import java.io.InputStreamReader;

fun main(){
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()
    val arr = Array<Node>(n){ Node(0,0) }

    for(i in 0..n-1){
        arr[i].value=br.readLine().toInt()
        arr[i].idx=i
    }

    arr.sort()

    var max = Integer.MIN_VALUE
    var index = 0
    arr.forEach{
        max = maxOf(max,it.idx-index++)
    }
    println(max+1)


}
class Node(var value:Int, var idx:Int) :Comparable<Node>{
    override fun compareTo(other: Node): Int {
        return value-other.value
    }


}